### PR TITLE
feat(surveys): add Rating question results viz

### PR DESCRIPTION
**Pull request by jurajmajerik**
Implemented a bar chart for Rating results as wireframed in https://github.com/PostHog/posthog/issues/17038

@corywatilo

Screenshot 2023-10-10 at 14 57 43

### CODE REVIEW

**Review by liyiy — APPROVED** [review]
Looks good!

Nonblocking: I think it'd be nice to add some sort of border or box to the graph so it has more structure to it on the page

File: frontend/src/scenes/surveys/surveyLogic.tsx [hunk 1]
```diff
@@ -172,6 +181,46 @@ export const surveyLogic = kea<surveyLogicType>([
                 }
             },
         },
+        surveyRatingResults: {
+            loadSurveyRatingResults: async ({
+                questionIndex,
+                question,
+            }: {
+                questionIndex: number
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyLogic.tsx:189]: `questionIndex` is sufficient here, since you can get the `question` from `values.survey.questions[questionIndex]`

File: frontend/src/scenes/surveys/surveyLogic.tsx [hunk 2]
```diff
@@ -172,6 +181,46 @@ export const surveyLogic = kea<surveyLogicType>([
...
+                }
+                const responseJSON = await api.query(query)
+                const { results } = responseJSON
+
+                const resultArr = new Array(question.scale).fill(0)
+                results?.forEach(([value, count]) => {
+                    resultArr[value - 1] = count
+                })
+
+                return { ...values.surveyRatingResults, [`question_${questionIndex}`]: resultArr }
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyLogic.tsx:221]: I think there's no good reason to have the keys be `question_X` instead of just the index. Makes things prone to mistyping errors with no extra benefit

Would recommend having an interface like:

```
export interface SurveyRatingResults {
    [key: number]: number[]
}

File: frontend/src/scenes/surveys/surveyViewViz.tsx [hunk 3]
```diff
@@ -119,20 +129,86 @@ export function Summary({
...
+}): JSX.Element {
+    const { loadSurveyRatingResults } = useActions(surveyLogic)
+
+    useEffect(() => {
+        loadSurveyRatingResults({ questionIndex, question })
+    }, [question])
+
+    return (
+        <div className="mb-4">
+            {!surveyRatingResultsReady[`question_${questionIndex}`] ? (
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyViewViz.tsx:169]: see, this is a lot more awkward than just:

```
surveyRatingResultsReady[questionIndex]
```

File: frontend/src/scenes/surveys/surveyViewViz.tsx [hunk 4]
```diff
@@ -119,20 +129,86 @@ export function Summary({
...
+                    )}
                 </>
             )}
         </div>
     )
 }
+
+export function RatingQuestionBarChart({
+    questionIndex,
+    question,
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyViewViz.tsx:152]: same idea as above, you don't need to pass in both questionIndex and question. Just the index is enough to uniquely identify the question.

And within the component you can get the question via `useValues(surveyLogic)`.

If this were in a backend context, I'd be all for trying to keep functions as pure as possible since there's no shared state. But here, there's already a lot of shared state, such that we run into the risk of mismatching the index and the question / having to always take care they are in sync when they represent the same data.

File: frontend/src/scenes/surveys/surveyViewViz.tsx [hunk 5]
```diff
@@ -92,12 +102,12 @@ export function UsersStackedBar({ surveyUserStats }: { surveyUserStats: SurveyUs
                     <div className="w-full flex justify-center">
                         <div className="flex items-center">
                             {[
-                                { count: seen, label: 'Viewed', color: 'bg-primary' },
-                                { count: dismissed, label: 'Dismissed', color: 'bg-warning' },
-                                { count: sent, label: 'Submitted', color: 'bg-success' },
-                            ].map(({ count, label, color }) => (
+                                { count: seen, label: 'Viewed', style: { backgroundColor: '#1D4AFF' } },
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyViewViz.tsx:105]: curious why this change? I think we're moving away from specific styles on components to utility classes for those styles (but not 100% sure)

**Review by neilkakkar — COMMENTED** [review]
Some nits that may or may not be useful for the future :)

File: frontend/src/scenes/surveys/surveyViewViz.tsx (hunk 5 above)
Comment (jurajmajerik) [reply frontend/src/scenes/surveys/surveyViewViz.tsx:105]: The colors suggested by Cory (https://github.com/PostHog/posthog/pull/17815) are different from those available in Tailwind.

Probably worth a separate PR to make them available there.
